kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/FEELFnResult.java

```diff
@@ -0,0 +1,22 @@
+package org.kie.dmn.feel.runtime.functions;
+
+import java.util.Optional;
+
+import org.kie.dmn.feel.runtime.events.FEELEvent;
+import org.kie.dmn.feel.util.Either;
+
+public class FEELFnResult<T> extends Either<FEELEvent, T> {
```

Comment:
@etirelli as discussed, I've implemented FEELFnResult<T> as an "alias" for Either<FEELEvent, T>.

This make the signature of the FEEL functions returning as `FEELFnResult<Object>`: at this point an Error is necessarily represented with a FEELEvent on the underlying Either-Left, while any valid result is represented by the underlying Either-right.

The `ofLeft()` has been named `ofError()` to mark it is necessarily an error, to be represented (not with an exception) with a FEELEvent.

The `ofRight()` has been named `ofResult()`.

I know there might be in the future cases where a "light" problem might occur, an we would like to raise a FEELEvent but still return an actual result - however, this imho should still be represented on the Either-right. This appears not be the case today, as all the actually present FEEL functions in case of any problem simply used to return a null.

what do you think?